src/routes/transactions/mappers/transaction-preview.mapper.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@nestjs/common';
+import * as winston from 'winston';
+import { MultisigTransaction } from '../../../domain/safe/entities/multisig-transaction.entity';
+import { Safe } from '../../../domain/safe/entities/safe.entity';
+import { DataDecodedService } from '../../data-decode/data-decoded.service';
+import { GetDataDecodedDto } from '../../data-decode/entities/get-data-decoded.dto.entity';
+import { PreviewTransactionDto } from '../entities/preview-transaction.dto.entity';
+import { TransactionPreview } from '../entities/transaction-preview.entity';
+import { TransactionDataMapper } from './common/transaction-data.mapper';
+import { MultisigTransactionInfoMapper } from './common/transaction-info.mapper';
+
+@Injectable()
+export class TransactionPreviewMapper {
+  constructor(
+    private readonly transactionInfoMapper: MultisigTransactionInfoMapper,
+    private readonly transactionDataMapper: TransactionDataMapper,
+    private readonly dataDecodedService: DataDecodedService,
+  ) {}
+
+  async mapTransactionPreview(
+    chainId: string,
+    safe: Safe,
+    previewTransactionDto: PreviewTransactionDto,
+  ): Promise<TransactionPreview> {
+    let dataDecoded;
+    try {
+      dataDecoded = await this.dataDecodedService.getDataDecoded(
+        chainId,
+        new GetDataDecodedDto(
+          previewTransactionDto.data,
+          previewTransactionDto.to,
+        ),
+      );
+    } catch (error) {
+      winston.warn(
+        `Error trying to decode the input data: ${JSON.stringify(error)}`,
```

Comment:
If there's an error, it should be a `DataSourceError`. I gave this a second thought and maybe we don't need to catch the error here since it would go through the error pipeline to the logs as any other error. This will simplify this function as well.

Reply:
Changed in [a158c2a](https://github.com/5afe/safe-client-gateway-nest/pull/291/commits/a158c2a02ceed6346e496ef948aac0185e20225d)